### DISCUSSION TITLE
Tooltip

### DISCUSSION THREAD

**Discussion by paur94** [recharts/recharts#3998] [Q&A]
Hello there,

I have a stacked bar chart, and I've created a custom tooltip with scrolling for handling large amounts of data. I want to ensure that if the tooltip is open and my mouse is on it, the tooltip doesn't change while I scroll through my data. Here's a video demonstrating the issue.

**Comment by ckifer** [comment]
Hi there

I'm a bit confused here

I see the custom tooltip ✅ 
I see the scrollbar in the tooltip ✅ 

I don't see you scrolling or if you are I cannot tell that you are. Where exactly is the issue in the video?

**Reply by paur94** [reply]
When hovering over a tooltip, I don't want it to affect the tooltips of other bars. I want to focus on the hovered tooltip until I scroll and can see all the data in that tooltip. Currently, when I hover over one bar and the tooltip is displayed, I can't see the tooltip's data properly because it changes when I move the mouse.

**Reply by ckifer** [reply]
I'm still confused - the tooltip is supposed to change when you move the mouse over another datapoint

Could you create a simple reproduction in codesandbox?
[![Edit stacked-bar-chart (forked)](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/stacked-bar-chart-forked-28ng4d?file=%2Fpackage.json)

**Reply by paur94** [reply]
Apologies for the miscommunication @ckifer. Allow me to provide a more detailed explanation.
I have a custom tooltip with a hover feature. In instances where I have a substantial amount of data, the tooltip includes a scroll function. However, when I attempt to move my mouse into the tooltip to scroll and view the data inside, the cursor movement inadvertently switches the tooltip to another bar's tooltip. How can I prevent the tooltip from closing (considering z-index or positioning) when my cursor is on it?

**Reply by ckifer** [reply]
Ah, thanks for the explanation that makes sense. 

Generally, I'd recommend against having scrollable content in your tooltip mostly because that much data in a small amount of space ends up not being viewed. 

Opinions aside, it may work if you change the trigger to click instead of hover? Currently there is no way to detect if the mouse is within the tooltip bounds so there won't be any way to prevent it from closing when it is

**Reply by paur94** [reply]
Thank you very much
